frontend/src/component/insights/componentsChart/useBatchedTooltipDate.ts

```diff
@@ -0,0 +1,33 @@
+import { useLocationSettings } from 'hooks/useLocationSettings';
+import { getDateFnsLocale } from '../getDateFnsLocale.ts';
+import type { TooltipItem } from 'chart.js';
+import { format, startOfWeek } from 'date-fns';
+
+export const useBatchedTooltipDate = () => {
+    const { locationSettings } = useLocationSettings();
+    const locale = getDateFnsLocale(locationSettings.locale);
+    return (datapoints: TooltipItem<any>[]) => {
+        const dataPoint = datapoints[0].raw as any;
+        if (
+            'date' in dataPoint &&
+            typeof dataPoint.date === 'string' &&
+            'endDate' in dataPoint &&
+            typeof dataPoint.endDate === 'string'
+        ) {
+            const startDate = format(
+                startOfWeek(new Date(dataPoint.date), {
+                    locale,
+                    weekStartsOn: 1,
+                }),
+                `PP`,
+                { locale },
+            );
+            const endDate = format(new Date(dataPoint.endDate), `PP`, {
+                locale,
+            });
+            return `${startDate} – ${endDate}`;
+        }
+
+        return undefined;
```

Comment:
According to [the chart.js docs](https://www.chartjs.org/docs/latest/configuration/tooltip.html#tooltip-callbacks):
> If the callback returns `undefined`, then the default callback will be used

So if we can't generate an appropriate label, we'll fall back to the default generation.